modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -489,10 +494,43 @@ void ONNXImporter::addNegation(const LayerParams& layerParams, opencv_onnx::Node
 
 void ONNXImporter::addConstant(const std::string& name, const Mat& blob)
 {
+    CV_LOG_DEBUG(NULL, "DNN/ONNX: add constant '" << name << "' shape=" << toString(shape(blob)) << ": " << toString(blob));
     constBlobs.insert(std::make_pair(name, blob));
     outShapes.insert(std::make_pair(name, shape(blob)));
 }
 
+void ONNXImporter::parseOperatorSet()
+{
+    int ir_version = model_proto.has_ir_version() ? (int)model_proto.ir_version() : -1;
+    if (ir_version < 3)
+        return;
+
+    int opset_size = model_proto.opset_import_size();
+    if (opset_size == 0)
+    {
+        CV_LOG_INFO(NULL, "DNN/ONNX: missing opset information")
+        return;
+    }
+
+    for (int i = 0; i < opset_size; ++i)
+    {
+        const ::opencv_onnx::OperatorSetIdProto& opset_entry = model_proto.opset_import(i);
+        const std::string& domain = opset_entry.has_domain() ? opset_entry.domain() : std::string();
+        int version = opset_entry.has_version() ? opset_entry.version() : -1;
+        if (domain.empty())  // ONNX opset
+        {
+            onnx_opset = std::max(onnx_opset, version);
+        }
+        else
+        {
+            // OpenCV don't know other opsets
+            CV_LOG_WARNING(NULL, "DNN/ONNX: unsupported opset[" << i << "]: name='" << domain << "' version=" << version);
```

Comment:
should we emit error here?

Reply:
I think we should change the condition to `domain.empty() || domain == "ai.onnx"` and emit the error if running normally, otherwise we should keep going(if `DNN_DIAGNOSTICS_RUN` is true).

Reply:
It is 3.4 branch, there is no `DNN_DIAGNOSTICS_RUN`